docs/source-2.0/tutorials/full-stack-tutorial.rst

```diff
@@ -0,0 +1,793 @@
+======================
+Full Stack Application
+======================
+
+Overview
+========
+In this tutorial, we’ll walk you through using Smithy in a full-stack application. This includes defining a model for
+a simple coffee-shop service, generating TypeScript code for both the client and server, and implementing both a
+front-end and back-end for the service. We won't assume that you're an expert in Smithy, but it may helpful to work
+through the :doc:`../quickstart` before beginning this tutorial.
+
+Let's get started.
+
+Setting Up the Project
+======================
+This application will consist of a 4 major components:
+
+1. The model
```

Comment:
could we add a brief description of each?

I.e. :  `the model - Smithy model representing the API of our Coffee Shop service`